static/app/components/nav/sidebar.tsx

```diff
@@ -89,37 +91,79 @@ const SidebarItemList = styled('ul')`
   }
 `;
 
-function SidebarItem({item}: {item: NavSidebarItem}) {
-  const location = useLocation();
-  const isActive = isNavItemActive(item, location);
-  const isSubmenuActive = isSubmenuItemActive(item, location);
-  const _to = resolveNavItemTo(item);
-  const linkProps = _to ? makeLinkPropsFromTo(_to) : {to: '#'};
+interface SidebarItemProps {
+  item: NavSidebarItem;
+}
+
+function SidebarItem({item}: SidebarItemProps) {
+  const to = resolveNavItemTo(item);
+  const Component = to ? SidebarLink : SidebarMenu;
```

Comment:
@JonasBa do you have any strong feelings on this pattern? It makes it a little trickier for typescript, vs if we did something like

```suggestion
  const item = to ? (
    <SidebarLink item={item} key={item.label} />
  ) : (
    <SidebarMenu item={item} key={item.label} />
  );
```

then typescript works a little less hard. In this case though they would need to be responsible for rendering their children

Reply:
It's not an uncommon pattern tbh. The drawback of this alternative is that the render body ends up bit less readable imo, as {item} doesnt convey that this is a component and could be any value. I tend to prefer to use this approach here, but if the type of component is limited I might just call it something else than `Component` to be more descriptive

```jsx
return (
	<div>
		{item}
		<Component ... />
   </div>
)
```

Reply:
Thanks for kicking off this conversation @evanpurkhiser! I agree with @JonasBa that the current `Component` pattern is well-suited here, since both component types expect identical props, children, and only differ in implementation. I did refactor to a more specific name in 9580e0c83f2. 

Would love to discuss possible TypeScript performance optimizations we can make across the codebase at the next FE TSC!